marlowe-playground-client/src/Simulation/BottomPanel.purs

```diff
@@ -0,0 +1,495 @@
+                                [ a_ [ text "Static Analysis" ] ]
+                            , li
+                                [ classes ([] <> isActive MarloweWarningsView)
+                                , onClick $ const $ Just $ ChangeSimulationView MarloweWarningsView
+                                ]
+                                [ a_ [ text $ "Warnings" <> if warnings == [] then "" else " (" <> show (length warnings) <> ")" ] ]
+                            , li
+                                [ classes ([] <> isActive MarloweErrorsView)
+                                , onClick $ const $ Just $ ChangeSimulationView MarloweErrorsView
+                                ]
+                                [ a_ [ text $ "Errors" <> if errors == [] then "" else " (" <> show (length errors) <> ")" ] ]
+                            ]
+                        , ul [ classes [ ClassName "end-item", aHorizontal ] ]
+                            [ li [ classes [ Classes.stateLabel ] ]
+                                [ text "Contract Expiration: ", state ^. (_marloweState <<< _Head <<< _contract <<< to contractMaxTime <<< to text) ]
+                            , li [ classes [ ClassName "space-left", Classes.stateLabel ] ]
+                                [ text "Current Blocks: ", state ^. (_marloweState <<< _Head <<< _slot <<< to show <<< to text) ]
+                            , li [ class_ (ClassName "space-left") ]
+                                [ a [ onClick $ const $ Just $ ShowBottomPanel (state ^. _showBottomPanel <<< to not) ]
+                                    [ img [ classes (minimizeIcon state), src closeDrawerIcon, alt "close drawer icon" ] ]
+                                ]
+                            ]
+                        ]
+                    ]
+                , panelContents state (state ^. _simulationBottomPanelView)
+                ]
+            ]
+        ]
+    ]
+  where
+  isActive view = if state ^. _simulationBottomPanelView <<< (to (eq view)) then [ ClassName "active-text" ] else []
+
+  warnings = state ^. (_marloweState <<< _Head <<< _editorWarnings)
+
+  errors = state ^. (_marloweState <<< _Head <<< _editorErrors)
+
+  hasRuntimeWarnings = state ^. (_marloweState <<< _Head <<< _transactionWarnings <<< to Array.null <<< to not)
+
+  hasRuntimeError = state ^. (_marloweState <<< _Head <<< _transactionError <<< to isJust)
+
+  contractMaxTime Nothing = "Closed"
+
+  contractMaxTime (Just contract) = let t = maxTime contract in if t == zero then "Closed" else show t
+
+panelContents :: forall p. FrontendState -> SimulationBottomPanelView -> HTML p HAction
+panelContents state CurrentStateView =
+  div [ class_ Classes.panelContents ]
+    [ div [ classes [ rTable, rTable6cols, ClassName "panel-table" ] ]
+        ( warningsRow <> errorRow
+            <> tableRow "Accounts" "No accounts have been used" "Account ID" "Participant" "Currency Symbol" "Token Name" "Money"
```

Comment:
Feels like this arguments are begging to be passed in the wrong order someday. Time for a record & explicit tuple?